### ISSUE TITLE
Question: Transaction rollback - No exception on context dispose?

### ISSUE THREAD

**Issue by HypeillingerChillinger** [linq2db/linq2db#3992]
In the release notes i found:
_disposal of transaction without explicit call to Rollback method will not call Rollback explicitly on transaction object anymore (transaction still be rolled back by database provider if needed)_ 

Does that mean, that on the disposal of the transaction the provider sends a rollback to the database but a possible exception will not be thrown by Linq2Db (which is good from my point of view)? Just want to be sure.

So for Postgres a disposal of a transaction will not throw the following `SqlState`s?

Class 40 — Transaction Rollback
40000 	transaction_rollback
40002 	transaction_integrity_constraint_violation
40001 	serialization_failure
40003 	statement_completion_unknown
40P01 	deadlock_detected

found here: https://www.postgresql.org/docs/current/errcodes-appendix.html

**Comment by MaceWindu** [comment]
It was a fix for situations, when transaction was closed (committed or rolled back) from code, not controlled by linq2db and then we called rollback on dispose. This resulted by `transaction not found` error from server because database provider wasn't perfroming any checks on transaction state.

Regarding your question: I cannot answer it as it depends on how Rollback and Dispose methods implemented in specific provider and you better ask it in Npgsql repo.
